### PR TITLE
Check OSXSAVE when detecting AVX support

### PR DESCRIPTION
**Pull request by 0dvictor**
OSXSAVE flag tells if OS enables XSAVE instruction,
which is a strong indication that AVX is enabled by
the OS.

Signed-off-by: Victor Ding <dvictor@ca.ibm.com>

### CODE REVIEW

**Comment by 0xdaryl** [comment]
These changes are only built on x86, so a genie build is not required.

File: compiler/x/runtime/X86Runtime.hpp [hunk 1]
```diff
@@ -66,6 +73,15 @@ inline bool jitGetCPUID(TR_X86CPUIDBuffer* pBuffer)
       // EAX = 7, ECX = 0
       cpuidex(CPUInfo, 7, 0);
       pBuffer->_featureFlags8 = CPUInfo[EBX];
+
+      // Check for XSAVE
+      if(pBuffer->_featureFlags2 & 0x08000000) // OSXSAVE
+         {
+         if((6 & _xgetbv(0) != 6) || feGetEnv("TR_DisableAVX")) // Unset OSXSAVE if not enabled via CR0
```
Comment (0xdaryl) [inline compiler/x/runtime/X86Runtime.hpp]: Can you annotate the naked constant `6` with a comment describing what XCR bits it represents?  e.g.,
```
// '6' = mask for XCR0[2:1]='11b' (XMM state and YMM state are enabled)
```

Comment (0dvictor) [reply compiler/x/runtime/X86Runtime.hpp]: Good suggestion, updated.

**Comment by 0xdaryl** [comment]
Thanks for the change.  Not sure if you saw my copyright comment above?

**Comment by 0dvictor** [comment]
Sorry I missed your comments about copyright headers. Now they are all updated.

**Pull request merged**